src/core/vmm.d

```diff
@@ -0,0 +1,489 @@
+module core.vmm;
+
+/**
+    $(P The module aims to provide the most commonly useful features of
+    target OS virtual memory manager subsystem. )
+
+    $(P The primary goal is not to provide full potential of every OS as
+        that is next impossible due to inherent differences.
+        Instead the goal is to cover $(STRONG the most common) scenarios
+        in an $(STRONG optimal) way. In particular issuing the absolute minimum
+        of system calls for each scenario.
+    )
+
+    Major use cases covered:
+    $(LI Getting platform's page size as cheap as reading a global. See $(LREF pageSize).
+    It takes a single system call at application start up.)
+    $(LI Allocating a memory block that has executable permission. A must have for JITs.
+    See $(LREF allocate) to allocate a block and set permissions in one call.)
+    $(LI Reserving a large virtual memory range and gradually committing memory from it.
+    Important for data-heavy applications. The permissions are set to read/write.
+    See $$(LREF reserve) and $(LREF commit).)
+    $(LI Locking a range of memory pages to persist in physical memory
+        (avoid being paging out by OS). See $(LREF lock) and $(LREF unlock).)
+
+    $(P API provides a way to perform less common operations but may have some overhead
+    compared to writing platform-specific code.)
+
+*/
+
+import core.thread;
+
+nothrow @nogc:
+
+/// Flags that specify desired kind of access to the memory.
+enum MemoryAccess : uint
+{
+    none =      0b0000_0000,
+    read =      0b0000_0001,
+    write =     0b0000_0010,
+    execute =   0b0000_0100,
+    readWrite = read | write,
+    readWriteExecute = read | write | execute,
```

Comment:
Worth noting that some operating systems (OpenBSD) or OS configurations (Linux+grsec) forbid this combination for security reasons, and perhaps we should discourage it as well by not providing a predefined constant for it. How about `readExecute` instead?